### DISCUSSION TITLE
Internal links to files and other notes inside Markdown notes

### DISCUSSION THREAD

**Discussion by sebastianjacobs** [standardnotes/forum#2931] [Suggestions]
I would like to have internal links to my notes and files I have uploaded to Standard Notes. So that I can embed images and link to other notes and files directly inside a note.

**Comment by shompoe** [comment]
Hi,    this feature is already available in the Supernote editor.    But that editor is still a work in progress.

**Comment by sebastianjacobs** [comment]
Nice! But I use markdown. Is this feature also planned for the markdown editors?

**Comment by shompoe** [comment]
Hi,

I am just a user like you, but gauging from everything I have ever read from the devs, I really doubt they will do much to change existing editors, which have been ported from other open source projects. But the Supernote editor seems to be their own implementation for better integration with the app.

Also, as native markdown doesn't support the features you are requesting, implementing those features in those editors would mean they wouldn't be markdown anymore. 
There is an option already in supernote to export the note contents back to markdown, albeit lacking anything not supported by markdown.
---


23 Dec 2022, 12:34 by ***@***.***:

**Comment by sebastianjacobs** [comment]
With internal urls this feature could be implemented without losing the markdown syntax.

For example in markdown the link would look like this:
`[Note title](standardnotes://458769487569456)`

These links wouldn't work outside Standard Notes and wouldn't behave like a normal link inside Standard Notes but you still have the markdown syntax.

They could also be added with an optional dropdown activated with `@`.

**Comment by jmason888** [comment]
+1 from another die-hard markdown user...

**Comment by myreli** [comment]
Hi @jmason888, I'm also a heavy Markdown user, and Super has been working fine with me. 

While its underlying system is not Markdown, it has support for the shortcuts, and you can export to MD as well. Since Markdown is just a formatting convention, but the file itself is plaintext, features like linking and embedding would not be feasible. 

What do you miss on super that you experience in a Markdown editor?

**Reply by sebastianjacobs** [reply]
I tested it. Super does not support all shortcuts and it transforms them directly into super components. I want to work directly in Markdown code without any fancy Super editor. Markdown is the **standard** for note taking and I hope it gets more support from **Standard** Notes (it's in the name).

It is not just the internal links I'm missing. I also miss code highlighting for example that has been added to super but not to Markdown editors: #2778 <-- see how nice it is to reference to something inside the text, GitHub uses markdown syntax too
